src/order/succ_pred/basic.lean

```diff
@@ -114,6 +114,29 @@ end preorder
 section linear_order
 variables [linear_order α]
 
+/-- A constructor for `succ_order α` for `α` a linear order. -/
+@[simps] def succ_order.of_core (succ : α → α) (hn : ∀ {a}, ¬ is_max a → ∀ b, a < b ↔ succ a ≤ b)
```

Comment:
I'm not sure what `simps` does here. It generates `succ_order.succ a = succ a`, but will it ever fire?

Reply:
```lean
example {a: fin 123}: succ_order.succ a = 123 :=
begin
  simp only [succ_order.of_core_succ],
  --ite (a < last (122.add 0)) (a + 1) a = 123
end
```

seems to fire; I wonder if this means I can get rid of the custom lemmas? (you also don't need to mention it by hand, try `simp` without any arguments)

Reply:
Hmm... okay. That's unexpected behavior to me, but if it works... I would still keep the custom lemmas for discoverability.

Reply:
Okay, sure, I'll do that.